MaraBot/Core/Preset.cs

```diff
@@ -1,14 +1,174 @@
+				if(Options.ContainsKey("opStatGrowth") && Options["opStatGrowth"] == "vanilla" && Options.ContainsKey("opDifficulty"))
+					return "Selecting a different difficulty is useless if you have vanilla enemy stat growth enabled. This is probably a mistake.";
+
+				// Use sensible mana seed settings
+				if((!Options.ContainsKey("opGoal") || Options["opGoal"] != "mtr") && (
+						Options.ContainsKey("opNumSeeds") ||
+						Options.ContainsKey("opMinSeeds") ||
+						Options.ContainsKey("opMaxSeeds")))
+					return "Using custom mana seed settings is useless if not enabling the Mana Tree Revival goal. This is probably a mistake.";
+				if(!Options.ContainsKey("opNumSeeds") && (Options.ContainsKey("opMinSeeds") || Options.ContainsKey("opMaxSeeds")))
+					return "Restricting the amount of mana seeds needed is useless if not using a random number of seeds needed. This is probably a mistake.";
+				if(Options.ContainsKey("opMinSeeds") && Options.ContainsKey("opMaxSeeds"))
+				{
+					// Comparison of string numbers works as expected in this case
+					int comp = String.Compare(Options["opMinSeeds"], Options["opMaxSeeds"]);
+					if(comp > 0)
+						return "Invalid 'mana seeds required' range.";
+					if(comp == 0)
+						return "'Mana seeds required' range is trivial. You should just set the 'Mana Tree Revival seeds required'.";
+				}
+
+				// Don't use the override christmas settings if already on a christmas goal
+				if(Options.ContainsKey("opGoal") && (Options["opGoal"] == "gift" || Options["opGoal"] == "reindeer"))
+				{
+					if(Options.ContainsKey("opXmasMaps"))
+						return "Setting the Christmas theme explicitly is unnecessary, as the goal is already Christmas-themed. This is probably a mistake.";
+					if(Options.ContainsKey("opXmasItems"))
+						return "Setting the random Christmas drops explicitly is unnecessary, as the goal is already Christmas-themed. This is probably a mistake.";
+				}
+			}
+
+			/*
+			 * Other
+			 */
+			foreach(var pair in Options)
+			{
+				// Skip if key doesn't exist
+				if(!Option.Options.ContainsKey(pair.Key))
+					continue;
+
+				var pairMode = Option.Options[pair.Key].Mode;
+
+				// Skip mode validation, as it's already been done
+				if(pairMode == Mode.Mode)
+					continue;
+
+				// All options must be general options or belong to the selected mode
+				if(pairMode != mode && pairMode != Mode.General)
+					return $"'{pair.Key}' belongs to the {pairMode.ToString()} mode, but the selected mode is {mode.ToString()}. This is probably a mistake.";
+
+				// Skip if value doesn't exist
+				if(!Option.Options[pair.Key].Values.ContainsKey(pair.Value))
+					continue;
+			}
+
+			return "";
+		}
+
+		/**
+		 * Parses the set of flags in Options to make a human readable list
+		 * of options, that can be filtered by Mode.
+		 */
+		public List<Tuple<Mode, string, string>> MakeDisplayable()
+		{
+			var err = ValidateOptions();
+			if(err != "")
+				throw new FormatException(err);
```

Comment:
I don't think we should validate options here. The bot should not need to validate if an option works for presets it's been loaded with.